samcli/lib/utils/file_observer.py

```diff
@@ -112,7 +112,8 @@ def _get_zip_lambda_function_paths(function_config: FunctionConfig) -> List[str]
             """
             code_paths = [function_config.code_abs_path]
             if function_config.layers:
-                code_paths += [layer.codeuri for layer in function_config.layers]
+                # Non-local layers will not have a codeuri property and don't need to be observed
```

Comment:
What is a "non-local" layer? If a Layer is in the template but does not have a CodeUri, shouldn't we fail instead of ignore? Not knowing where the layer code is, means we cannot add it to the container making testing not effective/confusing to customers. Usually we want the container to be as close to when the customer deploys as possible.

Reply:
By non-local, I mean a layer referencing a layer arn instead of a layer resource defined within the template. With these types of layers, we download them and they are still usable when running local invoke, start-api, etc. It only breaks when we try to watch the code source with --warn-containers. Since the source isn't on the user's machine there's nothing to watch, but the layer is still usable and testable.

Reply:
I see that makes sense. Should this be an integ tests to cover this case? Would have made understanding the change easier as well.

Reply:
I'll add an integration test.